keystore/filesystem/filesystem_backup.go

```diff
@@ -152,57 +165,155 @@ func readFilesAsKeys(files []string, basePath string, encryptor keystore2.KeyEnc
 				return nil, err
 			}
 		}
+
+		if isPublic(relativeName) {
+			if err := verifyPublicKey(&keys.PublicKey{
+				Value: content,
+			}); err != nil {
+				return nil, err
+			}
+		}
+
 		key := &keystore.Key{Name: relativeName, Content: content}
 		output = append(output, key)
 	}
 	return output, nil
 }
 
 // Export keys from KeyStore encrypted with new key for backup
-func (store *KeyBackuper) Export() (*keystore.KeysBackup, error) {
-	var publicKeys []*keystore.Key
+func (store *KeyBackuper) Export(exportIDs []keystore.ExportID, mode keystore.ExportMode) (*keystore.KeysBackup, error) {
+	var exportedKeys []*keystore.Key
 	var err error
-	if store.publicFolder != store.privateFolder {
-		publicFiles, err := ReadDir(store.storage, store.publicFolder)
-		if err != nil {
-			return nil, err
+
+	if len(exportIDs) != 0 {
+		for _, exportID := range exportIDs {
+			switch exportID.KeyKind {
```

Comment:
please, add comment that we additionally verify public keys and do not for private keys due to they are encrypted and will be validated on the decryption step. because in the future a question will arise as why so and do we need validate private keys too